src/sv-pipeline/scripts/make_evidence_qc_table.py

```diff
@@ -0,0 +1,264 @@
+    return df_ploidy
+
+
+def read_bincov_median(filename: str) -> pd.DataFrame:
+    """
+    Median coverage (bincov_median)
+    Args:
+        filename: bincov_median output from EvidenceQC (medianCov.transposed) in a tab separated format
+
+    Returns:
+        a pandas DataFrame containing the transposed table given in the input file.
+    """
+    df_median = pd.read_csv(filename, sep="\t").T
+    df_median = df_median.rename_axis(ID_COL).reset_index()
+    df_median.columns = [ID_COL, "median_coverage"]
+    df_median = df_median.reset_index(drop=True)
+    return df_median
+
+
+def read_wgd_scores(filename: str) -> pd.DataFrame:
+    return pd.read_csv(filename, sep="\t")
+
+
+def read_non_diploid(filename: str) -> pd.DataFrame:
+    """
+    EvidenceQC → untar → /ploidy_est/binwise_CNV_qValues.bed.gz →
+    count number of bins with q-value < 0.05 for each sample
+
+    Args:
+        filename: a tab-delimited file containing binwise CNV qValues.
+
+    Returns:
+        A pandas DataFrame containing the number of bins with q-value < 0.05 for each sample.
+    """
+    df_non_diploid = pd.read_csv(filename, sep="\t").T
+    df_non_diploid = df_non_diploid.T.iloc[0:, 3:]  # .rename_axis("#ID").reset_index()
+    new_c = df_non_diploid.iloc[0:, 0:]
+    nondiploid_counts = new_c[new_c < 0.05].count()
+    nondiploid_counts_df = pd.DataFrame(nondiploid_counts, columns=["nondiploid_counts"]).rename_axis(
+        "#ID").reset_index()
+    return nondiploid_counts_df
+
+
+def read_melt_insert_size(filename: str) -> array:
+    """
+    Reads MELT insert size into an array.
+    Args:
+        filename: melt_insert_size in sample data table
+
+    Returns:
+        Array[Float] of melt_insert_size for each sample
+    """
+    insert_size = pd.read_csv(filename, sep="\t")
+    melt_insert_size_array = pd.DataFrame(insert_size, columns=["melt_insert_size"]).astype(float)
+    return melt_insert_size_array
+
+
+def get_col_name(caller: str, outlier_type: str) -> str:
+    return f"{caller}_{outlier_type}_outlier"
+
+
+def read_outlier(filename: str, outlier_col_label: str, idx=10) -> pd.DataFrame:
+    """
+
+    Args:
+        filename: A tab-delimited file containing the output of
+        Evidence QC on a caller (e.g., Manta) with low/high.
+
+        outlier_col_label: Column header for the outliers.
+
+        idx: The index of the cell in the array containing sample
+        URI split by `/` that contains sample ID.
+    Returns:
+        A pandas DataFrame containing the number of times a sample
+        appears in the QC of a caller output.
+    """
+    df = pd.read_csv(filename, sep="\t")
+    if df["Outlier_Sample"].empty:
+        df[ID_COL] = df.apply(lambda _: "NA", axis=0)
+        outlier_sample = df.pivot_table(columns=[ID_COL], aggfunc="size").astype(int)
+    else:
+        df[ID_COL] = df["Outlier_Sample"].str.split("/", expand=True)[idx].str.split(".", expand=True)[0]
```

Comment:
I am not sure how extracting sample ID from its URI is portable/reliable. But I'll leave it to you to decide how best the sample ID can be extracted.

Reply:
@VJalili using `os.path.split()` could also be an option, but it was not working here.

Reply:
@shadizaheri do you have sample id in a different column of your data table? if so, that could be a better alternative, if not, it should be find to leave it as is.

Reply:
That makes sense, thanks!!